src/main/java/spleetwaise/transaction/logic/parser/AddCommandParser.java

```diff
@@ -50,21 +47,25 @@ private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Pre
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(
-                        args, PREFIX_PHONE, PREFIX_AMOUNT, PREFIX_DESCRIPTION, PREFIX_DATE, PREFIX_CATEGORY);
-        if (!arePrefixesPresent(argMultimap, PREFIX_PHONE, PREFIX_AMOUNT, PREFIX_DESCRIPTION)
-                || !argMultimap.getPreamble().isEmpty()) {
+                        args, PREFIX_AMOUNT, PREFIX_DESCRIPTION, PREFIX_DATE, PREFIX_CATEGORY);
+
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException e) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE), e);
+        }
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_AMOUNT, PREFIX_DESCRIPTION)) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE));
         }
-        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_PHONE, PREFIX_AMOUNT, PREFIX_DESCRIPTION, PREFIX_DATE);
-        Phone phone = spleetwaise.address.logic.parser.ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-        Person person = ParserUtil.getPersonFromPhone(phone);
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_AMOUNT, PREFIX_DESCRIPTION, PREFIX_DATE);
+
         Amount amount = ParserUtil.parseAmount(argMultimap.getValue(PREFIX_AMOUNT).get());
         Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get());
         Date date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).orElse(getNowDate()));
         Set<Category> categories = ParserUtil.parseCategories(argMultimap.getAllValues(PREFIX_CATEGORY));
 
-        Transaction transaction = new Transaction(person, amount, description, date, categories);
-
-        return new AddCommand(transaction);
+        return new AddCommand(index, amount, description, date, categories);
```

Comment:
I think the implementation here might have gone a bit too overboard. 

I do not think it would be necessary to break `AddCommand(transaction);` into `AddCommand(index, amount, description, date, categories)` specifying all the components. Doing so would **increase the coupling** since now if Parser changes, AddCommand would also have to change. 

Instead, I propose keeping the **original way of implementation** but instead of `Person person = ParserUtil.getPersonFromPhone(phone);`, it could be `Person person = ParserUtil.getPersonFromAddressBookList(index);` instead.

Reply:
@Dino-Nuggies, I do see that your current implementation has referenced the screenshot you provided in my comment for referencing how Index is used; just not so sure how this links to what @rollingpencil mentioned. Might also need you to clarify how Sean's previous comments conflict with the implementation that @rollingpencil suggested, seems to me that most data is still being handled in the AddCommand class.

My suggestion would be to refer to how `editTxn` is implemented, which, in this case, you can add an Index arg while retaining the other arg which is the Transaction object. The suggestion from @rollingpencil to use `ParserUtil.getPersonFromAddressBookList(index)` seems quite valid, I haven't looked into the details yet so I am not sure if there is any difficulty in not doing so.

```
EditCommand(Index index, EditPersonDescriptor editPersonDescriptor)
```